async-openai/src/error.rs

```diff
@@ -20,13 +24,35 @@ pub enum OpenAIError {
     FileReadError(String),
     /// Error on SSE streaming
     #[error("stream failed: {0}")]
-    StreamError(String),
+    StreamError(StreamError),
```

Comment:
Nice! Since you always map the inner eventsource error anyway, It would simplify the code quite a bit I think to wrap StreamError(EventSourceError) instead with #[from].

If I'm not mistaken, that would remove `map_stream_error` instead as we can use Into, remove the intermediate error, and avoid reading out the bytes of the response early.

Reply:
We can't because we need to determine what `OpenAIError` is wrapped inside the `EventSourceError::InvalidStatusCode` and as a bonus we also map the `EventSourceError::Transport` to `OpenAIError::Reqwest`.